version1/04 - Web services.md

```diff
@@ -1780,3 +1780,244 @@ Authorization: Bearer ACCESS_TOKEN
 **Response**
 
 204 No Content
+
+## Managing code list permissions
+
+This set of endpoints allows you to manage the explicit permissions on the specified objects:
+* all code lists
+* just the administrativEnhet code list or any of its list values
+* just the mappetype code list or any of its list values
+* just the dokumenttype code list or any of its list values
+* just the skjerming code list or any of its list values
+
+Address:
+```
+https://{server}:{port}/rms/api/public/noark5/v1/permission/code-list
+```
+
+### Get existing permissions
+
+Gets existing permissions for all code lists, a single code list, or a specific code list value.
+
+The use cases are, respectively:
+* GET /rms/api/public/noark5/v1/permission/code-list?listType=all
```

Comment:
Why not just `GET /rms/api/public/noark5/v1/permission/code-list` as in not specify the listType parameter?

Reply:
The underlying issue here is the fact that we don't treat code lists as resources in v1.

I was facing exactly the same challenge when I started on this and I was thinking exactly the same thing you are suggesting. However, we already have a `GET /code-lists` endpoint which has nothing to do with the parent of all lists and if I made this `GET /code-list`, I think this would have simply added to the problem, not solved it.

That's my reasoning and that's why I went with the documentation you're seeing now. I don't have any better suggestions either, but I decided that if I explicitly document the purpose of "all", it'd be better than treating just `code-list` as the parent of all lists.

The only thing that I could do (in the context of the already developed web services) is to probably add a "/code-list-parent" endpoint or the likes, but that seemed superfluous since I could simply create a new type of code list (all).